lib/ansible/parsing/vault/__init__.py

```diff
@@ -500,58 +503,42 @@ def rekey_file(self, filename, new_password):
         os.chown(filename, prev.st_uid, prev.st_gid)
 
     def read_data(self, filename):
+        data = None
 
-        try:
-            if filename == '-':
+        if filename == '-':
+            try:
                 data = sys.stdin.read()
-            else:
-                with open(filename, "rb") as fh:
-                    data = fh.read()
+            except Exception as e:
+                raise AnsibleError(str(e))
+            return
+
+        try:
+            with open(filename, "rb") as fh:
+                data = fh.read()
```

Comment:
my pref is to have the special case (filename='-') checked early and to return early and have the normal case be in the first indent level in the method. Also minimizes the amount of code that the raised AnsibleErrors could be referring to.